### ISSUE TITLE
Training on Conceptual Captions

### ISSUE THREAD

**Issue by ryntwn88** [rmokady/CLIP_prefix_caption#19]
Are there any important changes that need to be made to the train.py file to train a model with the the Conceptual Captions 3M dataset?

I've been attempting to train a model myself using the author's recommended settings for training an MLP with GPT2 finetune.  Hyperparameters such as learning rate, batch size, and num layers are all same as author.  My only changes have been upgrading the CLIP to ViT_B/16 and GPT2 to gpt2-medium.

Using the train.py script I am able to get my training to run and the reported loss is decreasing through the epochs. The problem comes when I run inference.  It always outputs "a model walks the runway at the fashion show during event." no matter what image I give it.

Any guidance would be appreciated.

**Comment by rmokady** [comment]
Hi @ryntwn88,
I would love to help but it would be hard to debug it without additional details

I'm not sure if you facing a bug in the inference script
For start I would add some debug logs to the training, dumping the the model outcome over the last 5 images or something like this. It would be interesting to see if the mode collapse occurring during the training itself

**Comment by ryntwn88** [comment]
@rmokady thank you for the reply and suggestion.  I added a step in the training loop to run inference on a set of hold out images. I also tested my inference script with your pretrained models as well as some of my own using the coco dataset to confirm the issue is not in my inference script.

I ran same training script in multiple configurations varying language models with gpt2-small and gpt2-medium and datasets with COCO training set, Conceptual Captions validation set and Conceptual Captions training set. This has narrowed down the problem to my Conceptual Captions training set as all other runs produce fairly accurate and meaningful captions.

Going to review if I have a bug in how I ran the parse_conceptual.py step

**Comment by ryntwn88** [comment]
Unable to find any bugs in the preprocessing, I started over fitting on small subsets of the training data to further validate there are no bugs in the pipeline. Got to observe the model adjusting through the training and am noticing it takes a while to break out of outputting the same (or very similar) captions on any given image. Started thinking i just need to train for more epochs. Attempted training for 20 epochs on 1m image-caption pairs from train set and ended with outputting same captions for all images.

Thinking it's time to start playing with the hyperparameters and see what happens.

**Comment by rmokady** [comment]
Which mapping network do you use? and do you fine-tune GPT?

I'm starting to believe there is but in our transformer code...

**Comment by ryntwn88** [comment]
I am using MLP mapping network and finetuning GPT

**Comment by rmokady** [comment]
Hi @ryntwn88,
I just rerun our model with MLP mapping network and finetuning GPT over Conceptual Captions
And it definitely output a different captions for different images.
So I don't see any issue with training nor inference code

**Comment by martinmclee** [comment]
> Unable to find any bugs in the preprocessing, I started over fitting on small subsets of the training data to further validate there are no bugs in the pipeline. Got to observe the model adjusting through the training and am noticing it takes a while to break out of outputting the same (or very similar) captions on any given image. Started thinking I just need to train for more epochs. Attempted training for 20 epochs on 1m image-caption pairs from train set and ended with a loss of 1.6 (should have mentioned in my initial attempts with the full training set the loss never got below 2.6) but still outputting same captions for all images.
> 
> Thinking it's time to start playing with the hyperparameters and see what happens.

Hi I've been experiencing exactly the same issue, may I know if you have found any clue about it?
